### ISSUE TITLE
Consider replacing requests

### ISSUE THREAD

**Issue by mahmoudhossam** [cjolowicz/hypermodern-python#188]
Requests has been the default suggestion for a python http client library, however it is starting to show its age.

I suggest replacing it in this guide with one of the alternatives, [httpx](https://www.python-httpx.org/)

It is a drop-in replacement for requests, and it also adds async support among other things.

**Comment by cjolowicz** [comment]
Agreed :) I'm working on the next version of the blog, and it does feature httpx instead of requests.

**Comment by wanderrful** [comment]
Out of curiosity, what do we get by using httpx over requests?

**Comment by mahmoudhossam** [comment]
The ability to make async requests.

Requests is synchronous by default, to make async requests you can either wrap it or use a requests plugin.
